core/src/main/java/tc/oc/pgm/community/features/VanishManager.java

```diff
@@ -0,0 +1,176 @@
+import tc.oc.pgm.community.events.PlayerVanishEvent;
+import tc.oc.pgm.listeners.ChatDispatcher;
+import tc.oc.pgm.listeners.PGMListener;
+import tc.oc.pgm.util.component.Component;
+import tc.oc.pgm.util.component.ComponentRenderers;
+import tc.oc.pgm.util.component.types.PersonalizedTranslatable;
+import tc.oc.pgm.util.named.NameStyle;
+
+public class VanishManager implements Listener {
+
+  private static final String VANISH_KEY = "isVanished";
+  private static final MetadataValue VANISH_VALUE = new FixedMetadataValue(PGM.get(), true);
+
+  private List<UUID> vanishedPlayers;
+  private final MatchManager matchManager;
+
+  public VanishManager(MatchManager matchManager) {
+    this.vanishedPlayers = Lists.newArrayList();
+    this.matchManager = matchManager;
+  }
+
+  public boolean isVanished(UUID uuid) {
+    return vanishedPlayers.contains(uuid);
+  }
+
+  private void addVanished(UUID uuid) {
+    if (!isVanished(uuid)) {
+      this.vanishedPlayers.add(uuid);
+    }
+  }
+
+  private void removeVanished(UUID uuid) {
+    this.vanishedPlayers.remove(uuid);
+  }
+
+  public List<MatchPlayer> getOnlineVanished() {
+    return vanishedPlayers.stream()
+        .filter(u -> matchManager.getPlayer(u) != null)
+        .map(matchManager::getPlayer)
+        .collect(Collectors.toList());
+  }
+
+  public void setVanished(MatchPlayer player, boolean vanish, boolean quiet)
+      throws CommandException {
+    if (isVanished(player.getId()) == vanish) {
+      throw new CommandException(
+          ComponentRenderers.toLegacyText(
+              new PersonalizedTranslatable(
+                  vanish ? "vanish.activate.already" : "vanish.deactivate.already"),
+              player.getBukkit()));
+    }
+
+    // Keep track of the UUID and apply/remove META data, so we can detect vanish status from other
+    // projects (i.e utils)
+    if (vanish) {
+      addVanished(player.getId());
+      player.getBukkit().setMetadata(VANISH_KEY, VANISH_VALUE);
+      player.getSettings().setValue(SettingKey.CHAT, SettingValue.CHAT_ADMIN);
+    } else {
+      removeVanished(player.getId());
+      player.getBukkit().removeMetadata(VANISH_KEY, VANISH_VALUE.getOwningPlugin());
+    }
+
+    final Match match = player.getMatch();
+
+    // Ensure player is an observer
+    if (!player.getParty().isObserving()) {
+      match.setParty(player, match.getDefaultParty());
+    }
+
+    // Set vanish status in match player
+    player.setVanished(vanish);
+
+    // Reset visibility to hide/show player
+    player.resetVisibility();
+
+    // Broadcast join/quit message
+    if (!quiet) {
+      PGMListener.announceJoinOrLeave(
+          player, vanish ? "broadcast.leaveMessage" : "broadcast.joinMessage");
+    }
+
+    match.callEvent(new PlayerVanishEvent(player, vanish));
+  }
+
+  /* Commands */
+
+  @Command(
+      aliases = {"vanish", "disappear", "v"},
+      desc = "Vanish from the server",
+      perms = Permissions.STAFF)
```

Comment:
I really dislike the approach of pgm.staff having all the perms, as it makes the plugin permissions a lot less customizable. I may want some staff groups in Stratus to be able to vanish but not freeze or lookup alts, especially with permission realms in mind, imagine a tournament setting for example.

I know it's down to @Electroid to decide in which parts of the project complexity is to be avoided, but I strongly believe that simplifying permission nodes is not worth the result.

Reply:
I think `pgm.vanish` should be ok.

Reply:
The issue if you make it `pgm.vanish` is that to setup permission for your staff you start getting a long list that becomes hard to manage. You end up doing permission node hunting to find out what to give and what not to give your mods access to, happened on 1.12 pgm and it became kinda hard to manage